File: app/src/main/java/ru/tsu/quizium/AuthorizationActivity.java
package ru.tsu.quizium;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class AuthorizationActivity extends AppCompatActivity {

    private FirebaseAuth mAuth;
    private FirebaseAuth.AuthStateListener mAuthListener;

    private EditText emailInputText;
    private EditText passwordInputText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_authorization);

        Button registerButton = findViewById(R.id.registerButton);
        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(AuthorizationActivity.this, RegistrationActivity.class);
                startActivity(intent);
            }
        });

        emailInputText = findViewById(R.id.emailLoginEditText);
        passwordInputText = findViewById(R.id.passwordLoginEditText);

        Button loginButton = findViewById(R.id.loginButton);
        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                signIng(emailInputText.getText().toString(), passwordInputText.getText().toString());
            }
        });
    }

    private void signIng(String email, String password) {
        mAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if(task.isSuccessful()) {
                            Toast.makeText(AuthorizationActivity.this, "Авторизация успешна", Toast.LENGTH_SHORT).show();
                            Intent intent = new Intent(AuthorizationActivity.this, MainActivity.class);
                            startActivity(intent);
                        } else {
                            Toast.makeText(AuthorizationActivity.this, "Не удалось авторизироваться", Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }

}

File: app/src/main/java/ru/tsu/quizium/AuthorizationActivity.kt
package ru.tsu.quizium

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.activity_authorization.*
import ru.tsu.quizium.dto.Result

class AuthorizationActivity : AppCompatActivity() {

    private val mAuth by lazy { FirebaseAuth.getInstance() }

    private val database by lazy { FirebaseDatabase.getInstance() }
    private val ref by lazy { database.reference }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_authorization)

        registerButton.setOnClickListener {
            val intent = Intent(this@AuthorizationActivity, RegistrationActivity::class.java)
            startActivity(intent)
        }


        loginButton.setOnClickListener {
            signIng(emailLoginEditText.text.toString(), passwordLoginEditText.text.toString())
        }
    }

    private fun signIng(email: String, password: String) {
        mAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(this) { task ->
                    if (task.isSuccessful) {

                        //val result = Result(1, 500)
                        //ref.child(mAuth.currentUser!!.uid).child(result.id.toString()).setValue(result)

                        Toast.makeText(this@AuthorizationActivity, "Авторизация успешна", Toast.LENGTH_SHORT).show()
                        val intent = Intent(this@AuthorizationActivity, MainActivity::class.java)
                        startActivity(intent)
                    } else {
                        Toast.makeText(this@AuthorizationActivity, "Не удалось авторизироваться", Toast.LENGTH_SHORT).show()
                    }
                }
    }
}